api/client/copy.go

```diff
@@ -0,0 +1,106 @@
+package client
+
+import (
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/docker/docker/pkg/archive"
+)
+
+// CmdCp handles copying files or directories to/from/accross containers.
+func (cli *DockerCli) CmdCp(args ...string) error {
+	cmd := cli.Subcmd("cp", "[SRCCONTAINER:]SRCPATH [DSTCONTAINER:]DSTPATH", "Copy files/folders from SRCPATH to the DSTPATH")
+	if err := cmd.Parse(args); err != nil {
+		return nil
+	}
+
+	if cmd.NArg() != 2 {
+		cmd.Usage()
+		return nil
+	}
+
+	var (
+		srcPath      = cmd.Arg(0)
+		dstPath      = cmd.Arg(1)
+		infoSrc      = strings.SplitN(srcPath, ":", 2)
+		infoDst      = strings.SplitN(dstPath, ":", 2)
+		srcContainer string
+		dstContainer string
+	)
+
+	if len(infoSrc) != 2 && len(infoDst) != 2 {
+		// User didn't specify any container.
+		return fmt.Errorf("error: Wrong path format. Must specify a source and/or destination container")
+	}
+
+	if len(infoSrc) == 2 {
+		// Copy from a container.
+		srcContainer, srcPath = infoSrc[0], infoSrc[1]
+	}
+	if len(infoDst) == 2 {
+		// Copy to a containers.
+		dstContainer, dstPath = infoDst[0], infoDst[1]
+	}
+
+	switch {
+	case dstContainer == "":
+		// Copy from a contaainer to a local path.
+		return cli.copyFromContainer(srcContainer, srcPath, dstPath)
+	case srcContainer == "":
+		// Copy from a local path to a container.
+		return cli.copyToContainer(srcPath, dstContainer, dstPath)
+	default:
+		// Copy from one container to another.
+		return cli.copyAccrossContainers(srcContainer, srcPath, dstContainer, dstPath)
+	}
+}
+
+func (cli *DockerCli) copyFromContainer(srcContainer, srcPath, dstPath string) error {
+	urlStr := fmt.Sprintf("/containers/%s/copy-from?path=%s", srcContainer, url.QueryEscape(srcPath))
+
+	response, statusCode, err := cli.call("GET", urlStr, nil, false)
+	if err != nil {
+		return err
+	}
+
+	if response != nil {
```

Comment:
I saw that the old CmdCp did this, but yeah, looking at the internals of `cli.call()` it looks like it's not possible.